src/Command/SystemLocaleCommand.php

```diff
@@ -0,0 +1,216 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Production\Command;
+
+use Doctrine\DBAL\Connection;
+use PDO;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Adapter\Console\ShopwareStyle;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class SystemLocaleCommand extends Command
+{
+    public static $defaultName = 'system:locale';
+
+    /**
+     * @var SymfonyStyle
+     */
+    protected $io;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        parent::__construct();
+        $this->connection = $connection;
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument('locale', InputArgument::REQUIRED, 'ISO locale for the shop');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $output = new ShopwareStyle($input, $output);
+        $output->section('Shop locale');
+
+        $locale = $input->getArgument('locale');
+        $this->setDefaultLanguage($locale);
```

Comment:
you definetly have to clear the cache after you changed the system language, 
the whole entity cache is stale at this point leading to numerous ugly bugs :laughing: 

**Been there, done that** :rofl:

Reply:
How do suggest to clear the cache? I thought I've done that here. I must admit I am a bit out of the topic by now :D

Reply:
You've done that when the command get's called as part of the system install command, not if it is run by itself.

You can inject the `\Shopware\Core\Framework\Adapter\Cache\CacheClearer` into the service and use the `clear()` method.

Reply:
Hey @JoshuaBehrens, could you take care of the comment from @keulinho? 🙂